docs/content/guides/dagster/ml-ops.mdx

```diff
@@ -0,0 +1,211 @@
+---
+title: Building machine learning pipelines In Dagster | Dagster Docs
+description: This guide illustrates how to use Dagster to operationalize your machine learning pipeline
+---
+
+# Building machine learning pipelines with Dagster
+
+In this guide, we’ll walk you through how to take your machine learning models and deploy and maintain them in production using Dagster, reliably and efficiently.
+
+We will work through building a machine lerning pipeline, including using assets for different elements, how to automate model training, and monitoring your model's drift.
+
+---
+
+## Why build your ML pipelines with Dagster?
+
+- Dagster makes iterating on machine learning models and testing easy and it is designed to use during the development process.
+- Dagster has a lightweight execution model means you can access the benefits of an orchestrator, like re-executing from the middle of a pipeline and parallelizing steps while you're experimenting.
+- Dagster models data assets, not just tasks, so it understands the upstream and downstream data dependencies.
+- Dagster is a one-stop shop for both the data transformations and the models that depend on the data transformations!
+
+<Note>
+  This guide assumes you have familiarity with machine learning concepts and
+  several Dagster concepts, including{" "}
+  <a href="/concepts/assets/software-defined-assets">software-defined assets</a>{" "}
+  and <a href="/concepts/ops-jobs-graphs/jobs">jobs</a> . We also recommend
+  reviewing the{" "}
+  <a href="/guides/dagster/automating-pipelines">
+    Automating Data Pipelines guide
+  </a>{" "}
+  as many of the same principles can be applied to machine learning.
+</Note>
+
+---
+
+## Machine learning development
+
+If you are already using Dagster for your ETL pipelines, it is a natural progression to build out and test your models in Dagster.
+
+For this guide, we will be using the Hacker News data demoed in the [tutorial](/tutorial).
+
+The machine learning model we will walk through takes the Hacker News stories and uses the titles to predict the number of comments that a story will generate. This will be a supervised model since we have the number of comments for all the previous stories.
+
+The assets graph will look like this at the end of this guide:
+
+<Image
+alt="alt"
+src="/images/guides/ml-ops/ml_asset_dag.png"
+width={1804}
+height={1064}
+/>
+
+### Ingesting Data
+
+First, we will create an asset that retrieves the most recent Hacker News records.
+
+```python file=/guides/dagster/ml_ops/ml_ops.py startafter=data_ingestion_start endbefore=data_ingestion_end
+import requests
+from dagster import asset, FreshnessPolicy
+import pandas as pd
+
+
+@asset(freshness_policy=FreshnessPolicy(maximum_lag_minutes=10))
```

Comment:
I think we should probably actually leave out the `FreshnessPolicy`, at least for now. Because we're not telling people how to schedule their model, so its presence is likely to be confusing.

I also think they might not necessarily want to update their training data every 10 minutes.

Reply:
ok removed from this and inference